sys-utils/setarch.c

```diff
@@ -293,6 +334,57 @@ static void verify_arch_domain(struct arch_domain *doms, struct arch_domain *tar
 	errx(EXIT_FAILURE, _("Kernel cannot set architecture to %s"), wanted);
 }
 
+static const struct { int value; const char * const name; } all_personalities[] = {
+#define X(opt) { .value = opt, .name = #opt },
+	ALL_PERSONALITIES
+#undef X
+};
+
+static const struct { int value; const char * const name; } all_options[] = {
+#define X(opt) { .value = opt, .name = #opt },
+	ALL_OPTIONS
+#undef X
+};
+
+static void show_current_personality(void)
+{
+	unsigned long options;
+	size_t i;
+	int pers = personality(0xffffffff);
+	if (pers == -1)
+		errx(EXIT_FAILURE, _("Can not get current kernel personality: %s"), strerror(errno));
+
+	options = pers & 0xffffff00;
+	pers &= 0xff;
+
+	for (i = 0; i < ARRAY_SIZE(all_personalities); i++) {
+		if (pers == all_personalities[i].value) {
+			printf("%s", all_personalities[i].name);
+			break;
+		}
+	}
+	if (i == ARRAY_SIZE(all_personalities))
+		printf("unknown:0x%x", pers);
```

Comment:
Not translated, use _(), and it would be probably better to be more verbose, something like _("unknown execution domains: 0x%x").

Reply:
Hm, I somehow don't really like to translate this output.
What about printing only the numeric value?

Reply:
Yes, printing for known flags, human-readable names, and 0x%x for the unknown is a good idea.